### PR TITLE
Fix bash script without conditional function

### PR DESCRIPTION
**Pull request by donnybeelo**
This is a compatibility improvement

The bash script that runs when ghostty is launched produces the following error:
```
bash: /home/dan/git/ghostty/zig-out/share/ghostty/shell-integration/bash/ghostty.bash: line 120: syntax error near unexpected token `('
bash: /home/dan/git/ghostty/zig-out/share/ghostty/shell-integration/bash/ghostty.bash: line 120: `        sudo() {'
```

This change should alleviate that by defining the original sudo function outside and aliasing sudo to that function. This should allow ghostty to work without issues on systems with sudo already aliased.

### CODE REVIEW

**Comment by tristan957** [comment]
cc @jparise

File: src/shell-integration/bash/ghostty.bash [hunk 1]
```diff
@@ -6,6 +6,26 @@
 # my simple bash usage this is working. If a bash expert wants to
 # improve this please do!
 
+sudo_function() {
```
Comment (jparise) [inline src/shell-integration/bash/ghostty.bash:9]: I don't think we want to leave this function in the global namespace. That was an advantage of the previous approach, along with only defining it when the `sudo` shell feature is enabled.

If we can't find another approach, we should rename this with a `__ghostty_` prefix like our other global functions.

File: src/shell-integration/bash/ghostty.bash [hunk 2]
```diff
@@ -117,25 +137,7 @@ function __ghostty_precmd() {
...
-              builtin break
-            fi
-          done
-          if [[ "$sudo_has_sudoedit_flags" == "yes" ]]; then
-            builtin command sudo "$@";
-          else
-            builtin command sudo TERMINFO="$TERMINFO" "$@";
-          fi
-        }
+        alias sudo="sudo_function"
```
Comment (jparise) [inline src/shell-integration/bash/ghostty.bash:140]: I'll also have to think a little bit about the differences between using an `alias` versus function override.

**Comment by jparise** [comment]
I thought about this a little more. I think we should also consider there two alternate approaches:

1. If `sudo` is already a function or an alias, skip our integration code.
2. If `sudo` is already a function or an alias, save a reference to that existing symbol, replace it with our function, and invoke that existing symbol instead of `builtin command sudo`. We can't assume compatibility, but users can always disable our 'sudo' shell feature in that case.

**Comment by mitchellh** [comment]
I like both approaches, so I'm open to either.

**Comment by donnybeelo** [comment]
It's more of an issue with bash checking for syntax errors in the ghostty script before it even runs, and erroring when it sees sudo being used as a function name when it exists as an alias. I implemented the second

**Comment by jparise** [comment]
I spent some more time looking at this and am proposing an alternate solution in #4080.

It uses the `function sudo` syntax, which won't conflict with an existing 'sudo' alias in a syntax-error-producing way.

**Pull request closed without merging**